cosmwasm/contracts/crosschain-registry/src/msg.rs

```diff
@@ -0,0 +1,139 @@
+    },
+
+    // Chain to Chain Channel Registry
+
+    // Set a source_chain->destination_chain->channel_id map in the registry
+    SetChainToChainChannelLink {
+        // The source chain
+        source_chain: String,
+        // The destination chain
+        destination_chain: String,
+        // The channel id
+        channel_id: String,
+    },
+    // Change an existing source_chain->destination_chain->channel_id map in the registry
+    ChangeChainToChainChannelLink {
+        // The source chain
+        source_chain: String,
+        // The destination chain
+        destination_chain: String,
+        // The new channel id
+        new_channel_id: String,
+    },
+    // Remove an existing source_chain->destination_chain->channel_id map in the registry
+    RemoveChainToChainChannelLink {
+        // The source chain
+        source_chain: String,
+        // The destination chain
+        destination_chain: String,
+    },
+
+    // Channel to Chain Chain Registry
+
+    // Set a channel_id->source_chain->destination_chain map in the registry
+    SetChannelToChainChainLink {
+        // The channel id
+        channel_id: String,
+        // The source chain
+        source_chain: String,
+        // The destination chain
+        destination_chain: String,
+    },
+    // Change an existing channel_id->source_chain->destination_chain map in the registry
+    ChangeChannelToChainChainLink {
+        // The channel id
+        channel_id: String,
+        // The source chain
+        source_chain: String,
+        // The new destination chain
+        new_destination_chain: String,
+    },
+    // Remove an existing channel_id->source_chain->destination_chain map in the registry
+    RemoveChannelToChainChainLink {
+        // The channel id
+        channel_id: String,
+        // The source chain
+        source_chain: String,
+    },
+
+    // Osmosis Denom Registry
+
+    // Set a native_denom->ibc_denom map in the registry
+    SetNativeDenomToIbcDenom {
+        // The native denom
+        native_denom: String,
+        // The ibc denom
+        ibc_denom: String,
+    },
+
+    // Change an existing native_denom->ibc_denom map in the registry
+    ChangeNativeDenomToIbcDenom {
+        // The native denom
+        native_denom: String,
+        // The new ibc denom
+        new_ibc_denom: String,
+    },
+
+    // Remove an existing native_denom->ibc_denom map in the registry
+    RemoveNativeDenomToIbcDenom {
+        // The native denom
+        native_denom: String,
+    },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(GetAddressFromAliasResponse)]
+    GetAddressFromAlias { contract_alias: String },
+
+    #[returns(GetChainToChainChannelLinkResponse)]
+    GetChainToChainChannelLink {
+        source_chain: String,
+        destination_chain: String,
+    },
+
+    #[returns(crate::helpers::QueryDenomTraceResponse)]
+    GetDenomTrace { hash: String },
```

Comment:
Is the idea here that we "reexport" the denom trace query so that other contracts can use it without having to define the prost structs?

Reply:
Yes! Was hoping this would just be the one stop shop for all info/queries CXS requires